Single cell representation learning

agitter opened the pull request:
I made a first pass at adding #346 (and #647, though I haven't looked at it closely) to the single cell section.  @gwaygenomics can you please review?  We should better connect these methods to your work and others on bulk expression data, but I didn't have time for that now.  I took out the line about image classification because it no longer seemed relevant to me.

@zivbj and @tmfs10 can you please see if this is an accurate summary?  I was intrigued by the retrieval performance in different cell types and whether that is a biological property or related to the abundance of various cell types in the training data.

@akundaje I welcome your comments too.

akundaje commented:
@agitter I added some more clarifications to my comments and Ziv's response here. https://github.com/greenelab/deep-review/issues/346#issuecomment-326751627 . I am wondering what you think. Are my concerns valid or am I missing something.

agitter commented:
@akundaje I'll take the discussion back to #346 since your main comments are there.  I'll refrain from merging this until we hear from Sid and Ziv and have a consensus about what to say.

gwaygenomics commented on sections/04_study.md:
```diff
@@ -525,9 +525,11 @@ types where the subset percentage ranged from 0.1 to 1%, significantly
-identification, but the stunning improvements in image classification over the
-past 5 years [@tag:He2015_images] suggest transformative potential.
+
+Neural networks can also learn meaningful low dimensional representations of single cell expression data [@doi:10.1093/nar/gkx681 @doi:10.1101/178624], which can be used to identify cellular subpopulations or match new cells to known cell types.
```
I read the [Ding et al.](http://www.biorxiv.org/content/early/2017/09/01/178624) paper yesterday. The paper is about a deep VAE trained to embed a 2 dimensional hidden layer. It's used primarily as a visualization tool that the authors demonstrate out-compete PCA/t-SNE in 1) preserving local distances in the presence of outlier cells and 2) projecting new data into 2D space. They do not go into much detail about identifying subpopulations or matching known cell types. I suggest reworking the citation to this paper

evancofer replied:
For reference, [Lopez et al.](https://arxiv.org/abs/1710.05086) also examines the use of VAEs for scRNA-seq, but does more analysis than just visualization. It considers specific use cases (e.g., differential expression, imputing missing data) as well.

agitter replied:
@evancofer thanks, I hadn't seen that one.  I'd like to merge this pull request soon so I'll check it out.  I created #682.

agitter commented:
I'm going to wrap this up.  For now, I merged with master and added a placeholder for #682.  I still haven't addressed the comments above.

@gwaygenomics I'll let you know when I'm ready for a re-review.

agitter commented:
@gwaygenomics and @evancofer can you please review this updated version?  I didn't have much to say about the VAEs.  scVI looks interesting, but the results are quite brief in this version of the manuscript.

gwaygenomics commented on content/04.study.md:
```diff
@@ -525,9 +525,13 @@ types where the subset percentage ranged from 0.1 to 1%, significantly
-past 5 years [@tag:He2015_images] suggest transformative potential.
+
+Neural networks can also learn low-dimensional representations of single-cell gene expression data for visualization, clustering, and other tasks.
+Both scvis [@doi:10.1101/178624] and scVI [@arxiv:1709.02082] are unsupervised approaches based on VAEs.
```
It may be worth adding one or two sentences here describing the differences between the two papers - they have different goals.

`scvis` is primarily for visualization as it compresses input data into 2 or 3 dimensions. The authors determine that their model preserves local distances better than t-sne.

`scVI` is presented more as a general modeling approach and introduces a custom zero-inflated loss into the VAE objective.

agitter replied:
@gwaygenomics does be27965 work better?  The sentence is long, but it was awkward to write because I can't start a new sentence with the lower-case method names.

gwaygenomics replied:
:+1:

agitter commented:
Ziv is going to look at this section but can't today.  Let's wait to merge.

@gwaygenomics I'll incorporate your suggestions.

agitter commented:
I'm going to merge now that @gwaygenomics approved.

@zivbj, you can still make comments here if you have any.  I'll make any new edits in a separate pull request.  We're trying to close all pull requests to do a global reflow of the manuscript.

@dhimmel is Greycite down?  The new t-SNE reference isn't in the cache and the [Greycite request failed](https://travis-ci.org/greenelab/deep-review/builds/318711904#L776).  I made a note in #699 to fix this.

dhimmel commented:
> is Greycite down? 

it's been down for a while. will look into workarounds
